### PR TITLE
Extending the schema inference tool.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

**Comment by dkorolev** [comment]
@mzhurovich -- further extended the tool to:

* Preserve the order of fields when inferring `CURRENT_STRUCT`s from JSONs.
* Preserve the order of fields when converting from [T|C]SV into JSON.
* Convert integers into integers and doubles into doubles when converting from [T|C]SV into JSON.
* Added two scripts to do "all the magic" for a given `X.tsv` or `X.csv` files respectively.

Thanks,
Dima

File: Utils/JSONSchema/golden/multiple_objects.schema [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   CURRENT_FIELD(string, std::string);
+  CURRENT_FIELD(maybe, bool);
 };
 
 CURRENT_STRUCT(Schema_Object) {
-  CURRENT_FIELD(b, bool);
-  CURRENT_FIELD(optional, Optional<std::vector<Schema_optional_Element_Object>>);
   CURRENT_FIELD(s, std::string);
+  CURRENT_FIELD(b, bool);
+  CURRENT_FIELD(optional, std::vector<Schema_optional_Element_Object>);
```
Comment (mzhurovich) [inline Utils/JSONSchema/golden/multiple_objects.schema]: What's happened to `Optional`s?

Comment (dkorolev) [inline Utils/JSONSchema/golden/multiple_objects.schema]: Они утонули.

Good catch, :+1:, d0a7f34, PTAL.

**Comment by mzhurovich** [comment]
@dkorolev 👍

**Pull request merged**